cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -308,6 +308,17 @@ func (m *AwsManager) buildNodeFromTemplate(asg *asg, template *asgTemplate) (*ap
 			node.Spec.Taints = append(node.Spec.Taints, mngTaints...)
 			klog.V(5).Infof("node.Spec.Taints : %+v\n", node.Spec.Taints)
 		}
+
+		mngTags, err := m.managedNodegroupCache.getManagedNodegroupTags(nodegroupName, clusterName)
+		if err != nil {
+			klog.Errorf("Failed to get tags from EKS DescribeNodegroup API for nodegroup %s in cluster %s because %s.", nodegroupName, clusterName, err)
+		} else if mngTags != nil && len(mngTags) > 0 {
+			resourcesFromTags := extractAllocatableResourcesFromTags(mngTags)
```

Comment:
This is shadowing the variable `resourcesFromTags` defined on line 275. That variable is used to populate resource capacities on lines 277-279 from the ASG template. I gather from your PR summary that you intend specifically for the values from the NodeGroupTags to override any values previously set by that code. Perhaps worth adding a code comment here indicating that if there is a Managed Node Group then its tags will override anything from an ASG? I think? :) Honestly, it's really not clear to me how Managed Node Groups are different from "normal" EC2 ASGs...

Reply:
You are right, `resourcesFromTags` probably shouldn't be shadows so I renamed this variable to `resourcesFromMngTags`.
I also added a comment making it clear that if the same tag with different values were present on the ASG and the MNG the value on the MNG will take precedence.